modules/pam_succeed_if/pam_succeed_if.c

```diff
@@ -350,6 +350,18 @@ evaluate(pam_handle_t *pamh, int debug,
 			tty == NULL)
 			tty = "";
 		left = (const char *)tty;
+	} else if (strncasecmp(left, "env:", 4) == 0 && strlen (left + 3) > 1) {
+		const char *env = left + 4;
+		left = pam_getenv(pamh, env);
+		if (strcasecmp(qual, "is") == 0) {
+			if (strcasecmp(right, "set") == 0) {
+				return left != NULL ? PAM_SUCCESS : PAM_AUTH_ERR;
+			}
+			if (strcasecmp(right, "unset") == 0) {
+				return left == NULL ? PAM_SUCCESS : PAM_AUTH_ERR;
+			}
+			qual = "=";
```

Comment:
Please make sure this won't segfault in case of `left == NULL`.
Also, please make sure operations make sense when the variable is not set.